runtime/expr.h

```diff
@@ -503,32 +503,29 @@ class let_stmt : public stmt_node<let_stmt> {
   static constexpr node_type static_type = node_type::let_stmt;
 };
 
-// A block is a set of two other `stmt`s. Recursively using blocks allows an arbitrarily large set of `stmt`s to be
-// defined in sequence.
+// A block is a list of (generally) two or more `stmts`.
+//
+// (TODO: a block with a single stmt should be uncommon but not sure if it should be illegal.)
```

Comment:
I think this comment is unnecessary, the API doesn't let you do any of these things anyways.

Instead, I'd comment on `block::make` that it flattens nested blocks, and removes undef stmts from the vector.